portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -11,31 +12,23 @@
 
 @WebServlet("/login-status")
 public class LoginServlet extends HttpServlet {
+  /* Struct that holds the user's login state */
+  private class UserStatus {
+    boolean logged;
+    String url;
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     UserService userService = UserServiceFactory.getUserService();
-    boolean loggedIn = userService.isUserLoggedIn();
-    String url = loggedIn ? userService.createLogoutURL("/") : userService.createLoginURL("/");
 
-    response.setContentType("application/json;");
-    response.getWriter().println(toJson(loggedIn, url));
-  }
+    UserStatus userStatus = new UserStatus();
```

Comment:
nit: you can have a constructor `UserStatus(boolean logged, String url)` that initialises the fields directly, instead of setting them outside.